src/main/kotlin/graphics/scenery/controls/SwingUiNode.kt

```diff
@@ -0,0 +1,100 @@
+package graphics.scenery.controls
+
+import graphics.scenery.attribute.material.Material
+import graphics.scenery.primitives.Plane
+import graphics.scenery.textures.Texture
+import graphics.scenery.utils.Image
+import org.joml.Matrix4f
+import org.joml.Vector2f
+import org.joml.Vector3f
+import org.joml.Vector4f
+import java.awt.Color
+import java.io.FileInputStream
+import java.io.InputStream
+
+/**
+ * @author Jan Tiemann
+ * @author Konrad Michel
+ *
+ * Scene representation of a SwingFrame on a plane in form of a texture snapshot render of the frame in combination with SwingBridgeFrame
+ */
+class SwingUiNode(val swingBridgeFrame : SwingBridgeFrame) : Plane(
+    Vector3f(-0.5f,-0.5f,0.0f),
+    Vector3f(-0.5f,0.5f,0.0f),
+    Vector3f(0.5f,-0.5f,0.0f),
+    Vector3f(0.5f,0.5f,0.0f)
+) {
+    var swingUiDimension = 0 to 0
+
+    init {
+        //to also hit the backside
+        this.material().cullingMode = Material.CullingMode.None
+        /*this.material {
+            textures["diffuse"] = Texture.fromImage(Image.fromStream(FileInputStream("C:\\Users\\Kodels Bier\\Desktop\\volumes\\helix.png"), ".png", false))
+        }*/
+        this.update += {
+            this.spatial().scale = Vector3f(swingBridgeFrame.width.toFloat()/swingBridgeFrame.height.toFloat(), 1.0f, 1.0f)
+            spatial().needsUpdate = true
+        }
+    }
+
+    /**
+     * Function that takes the last taken screenshot from the SwingUI-window coming from the attached swingBridgeFrame and updates the SwingUINodes-Plane texture with it
+     */
+    fun updateUITexture() {
+        swingBridgeFrame.finalImage?.let {
+            this.material {
+                textures["diffuse"] = Texture.fromImage((it), Texture.RepeatMode.Repeat.all(), Texture.BorderColor.OpaqueBlack, true, true,
+                    Texture.FilteringMode.NearestNeighbour, Texture.FilteringMode.NearestNeighbour, hashSetOf(Texture.UsageType.Texture))
+            }
+        }
+    }
+
+    fun Matrix4f.copy(): Matrix4f = Matrix4f(this)
```

Comment:
This is only used once and will pollute the namespace if this class is imported.